### PR TITLE
Clear password

### PR DESCRIPTION
**Pull request by wussler**

### CODE REVIEW

File: bcrypt.go [hunk 1]
```diff
@@ -122,12 +123,20 @@ func HashBytes(password []byte, salt ...[]byte) (hash []byte, err error) {
...
 	sr := bytes.NewBuffer(s)
-	if !consume(sr, '$') || !consume(sr, '2') || !consume(sr, 'y') || !consume(sr, '$') {
+
+	if !consume(sr, '$') || !consume(sr, '2') {
 		return nil, InvalidSalt
 	}
 
+	if !consume(sr, '$') {
+		minor, _ = sr.ReadByte()
+		if minor != 'a' || !consume(sr, '$') {
```
Comment (twiss) [inline bcrypt.go]: Rather than reverting it in this PR (since this is kind of a hidden breaking change), I would make a separate PR which reverts this commit, _and_ does something like:
```suggestion
		if (minor != 'a' && minor != 'y') || !consume(sr, '$') {
```
That way, it's not a breaking change, it fixes internally generated salts, _and_ we can pass `$2y$` and then we don't have to modify the returned hash.

File: bcrypt.go [hunk 2]
```diff
@@ -161,10 +170,11 @@ func HashBytes(password []byte, salt ...[]byte) (hash []byte, err error) {
 	}
 
 	// cipher expects null terminated input (go initializes everything with zero values so this works)
-	password_term := make([]byte, len(password)+1)
-	copy(password_term, password)
+	passwordTerm := make([]byte, len(password)+1)
+	copy(passwordTerm, password)
```
Comment (twiss) [inline bcrypt.go]: This is very nitpicky but, since the entire rest of this module uses snake_case, I wouldn't change it for this one variable either.

File: bcrypt.go [hunk 3]
```diff
@@ -179,3 +189,9 @@ func MatchBytes(password []byte, hash []byte) bool {
 	}
 	return subtle.ConstantTimeCompare(h, hash) == 1
 }
+
+func clear(w []byte) {
+	for k := range w {
+		w[k] = 0x00
+	}
+}
```
Comment (twiss) [inline bcrypt.go]:
```suggestion
		w[k] = 0
	}
}
```
GitHub complains about this missing newline - not super sure if this suggestion will fix it

**Review by twiss — COMMENTED** [review]
LGTM :+1: Some nit picks below

File: bcrypt.go (hunk 3 above)
Comment (wussler) [reply bcrypt.go]: I added the newline from the editor

**Comment by twiss** [comment]
Merged as bbdd3ea727a4dd0ad7a376d89d0332bfdf90aba8 and 227b4adcab57223f6dd44a27c8ccac50c54090d9, respectively.
